openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -775,13 +799,397 @@ function scale(IndicatorObj) {
+            return;
+        }
+    }
+
+    // Focus map on layer
+    if (projectChange) {
+        map.fitBounds(mapboxBoundingBox);
+    }
+
+    var min = Math.min.apply(null, minMaxArray).toFixed(2);
+    var max = Math.max.apply(null, minMaxArray).toFixed(2);
+    min = (parseFloat(min) - 0.1);
+    // Round up the max
+    max = Math.ceil(max * 10) / 10;
+
+    var breaks = [];
+
+    function getColor() {
+        var interval = (max - min) / 6;
+        var tempStep = min;
+        for (var i = 0; i < 5; i++) {
+            tempStep += interval
+            breaks.push(tempStep);
+        }
+        breaks.unshift(min);
+        breaks.push(max);
+    }
+
+    getColor();
+
+    // Case 2, and 4, try to remove any existing layers
+    // TEMP also case 3
+    //if (weightChange === false) {
+        try {
+            for (var i = 0; i < 6; i++) {
+                map.removeLayer(i);
+            };
+        } catch (e) {
+            // continue
+        }
+    //}
+
+    // Case 3 update the source
+    if (weightChange) {
+        //console.log('Update the source:');
+        //map.update(true);
+
+        // FIXME would really prefer to update the map source here
+        map.removeSource('projectSource');
+    }
+
+    // Case 4 destory the source
+    // Here we use a try catch to manage 2 cases:
+    // 1. When a project is loaded for the first time. In this case there is no existing map source or layer,
+    // so we try to remove them, but as they do not exist, the app moves on and builds a new source and layer.
+    // 2 When a project is loaded for the 2nd, 3rd, nth time. In this case we find that there is a map source
+    // and layer that need to be removed, after being removed, the new map source and layers are created.
+    // The same logic applies to removeing layers.
+    if (projectChange) {
+        try {
+            map.removeSource('projectSource');
+        } catch (e) {
+            // continue
+        }
+    }
+
+    // Case 1 and 4
+    // **TEMP** also case 3
+    // Populate the mapbox source with GeoJson from Geoserver
+    // Only create a new source when the project has been changed
+    if (map.style.sources.projectSource === undefined || projectChange || weightChange) {
+        map.addSource('projectSource', {
+            'type': 'geojson',
+            'data': mappingLayerAttributes,
+        });
+        projectChange = false;
+    }
+
+    $('#mapLegend').remove();
+    // Create the map legend
+    $('#map-tools').append(
+        '<div id="mapLegend" class="my-legend">'+
+            '<div class="legend-title">'+selectedIndicator+'</div>'+
+            '<div class="legend-scale">'+
+                '<ul id="legendLables" class="legend-labels">'+
+                '</ul>'+
+            '</div>'+
+        '</div>'
+    );
+
+    // Cases 1, 2, 3, 4, and 5
+    //if (weightChange === false) {
+        // Create a new mapbox layers
+        for (var i = 0; i < 6; i++) {
```

Comment:
Please check about the commented code and see if possibly deleting it and re-indenting the inner block